base/ca/src/main/java/org/dogtagpki/server/ca/CAEngine.java

```diff
@@ -186,10 +188,11 @@ public class CAEngine extends CMSEngine {
 
     public CAEngine() {
         super("CA");
+        instance = this;
     }
 
     public static CAEngine getInstance() {
-        return (CAEngine) CMS.getCMSEngine();
+        return instance;
```

Comment:
The instance should be set here if `null` and removed from the constructor for a better design

Reply:
@edewata I do not think it is better to modify the `CAWebListener`. Since the current implementation requires to get the null value than I would prefer to merge as it is now and eventually improve the design  in future update.